app/routes.py

```diff
@@ -351,54 +361,105 @@ def add_harvest_source():
+                "data": [job.records_errored for job in jobs],
+                "borderColor": "red",
+                "backgroundColor": "red",
+            },
+            {
+                "label": "Ignored",
+                "data": [job.records_ignored for job in jobs],
+                "borderColor": "grey",
+                "backgroundColor": "grey",
+            },
+        ],
+    }
+    data = {
+        "harvest_source": source,
+        "harvest_source_dict": db._to_dict(source),
+        "harvest_jobs": jobs,
+        "chart": chartdata,
+        "next_job": next_job,
+    }
+    return render_template("view_source_data.html", data=data)
+
+
 @mod.route("/harvest_source/", methods=["GET"])
-@mod.route("/harvest_source/<source_id>", methods=["GET", "POST"])
-def get_harvest_source(source_id: str = None):
-    if source_id:
-        source = db._to_dict(db.get_harvest_source(source_id))
-        if request.args.get("type") and request.args.get("type") == "json":
-            return source
-        return render_template(
-            "view_data.html",
-            data=source,
-            action="View",
-            data_type="Harvest Source",
-            source_id=source_id,
-        )
-    else:
-        source = db.get_all_harvest_sources()
-        return db._to_dict(source)
+@mod.route("/harvest_sources/", methods=["GET"])
+def view_harvest_sources():
+    sources = db.get_all_harvest_sources()
+    data = {"harvest_sources": sources}
+    return render_template("view_source_list.html", data=data)
 
 
 # Edit Source
-@mod.route("/harvest_source/edit/<source_id>", methods=["GET", "POST"])
+@mod.route("/harvest_source/config/edit/<source_id>", methods=["GET", "POST"])
 @login_required
-def edit_harvest_source(source_id: str = None):
+def edit_harvest_source(source_id: str):
     if source_id:
-        source = db._to_dict(db.get_harvest_source(source_id))
-        organizations = db._to_dict(db.get_all_organizations())
-        organization_choices = [
-            (str(org["id"]), f'{org["name"]} - {org["id"]}') for org in organizations
-        ]
-        form = HarvestSourceForm(data=source)
-        form.organization_id.choices = organization_choices
-        if form.validate_on_submit():
-            new_source_data = make_new_source_contract(form)
-            source = db._to_dict(db.update_harvest_source(source_id, new_source_data))
-            if source:
-                flash(f"Updated source with ID: {source['id']}")
-            else:
-                flash("Failed to update harvest source.")
-            return redirect(f"/harvest_source/{source['id']}")
-        return render_template(
-            "edit_data.html",
-            form=form,
-            action="Edit",
-            data_type="Harvest Source",
-            button="Update",
-            source_id=source_id,
-        )
+        source = db.get_harvest_source(source_id)
+        organizations = db.get_all_organizations()
```

Comment:
we have 133. that number probably isn't going to increase enough to make this call expensive. just a comment.